openedx/features/content_tagging/api.py

```diff
@@ -0,0 +1,135 @@
+from organizations.models import Organization
+
+from .models import ContentTag, ContentTaxonomy, TaxonomyOrg
+
+
+def create_taxonomy(
+    name: str,
+    org_owners: List[Organization] = None,
+    description: str = None,
+    enabled=True,
+    required=False,
+    allow_multiple=False,
+    allow_free_text=False,
+    taxonomy_class: Type = ContentTaxonomy,
+) -> Taxonomy:
+    """
+    Creates, saves, and returns a new Taxonomy with the given attributes.
+
+    If `taxonomy_class` not provided, then uses ContentTaxonomy.
+
+    If `org_owners` is empty/None, then the returned taxonomy is enabled for all organizations.
+    """
+    taxonomy = oel_tagging.create_taxonomy(
+        name=name,
+        description=description,
+        enabled=enabled,
+        required=required,
+        allow_multiple=allow_multiple,
+        allow_free_text=allow_free_text,
+        taxonomy_class=taxonomy_class,
+    ).cast()
+    if org_owners:
+        set_taxonomy_org_owners(taxonomy, org_owners)
+    return taxonomy
+
+
+def set_taxonomy_org_owners(
+    taxonomy: Taxonomy,
+    org_owners: List[Organization] = None,
+):
+    """
+    Updates the list of org "owners" on the given taxonomy.
+
+    If `org_owners` is empty/None, then the taxonomy is enabled for all organizations.
+    """
+    TaxonomyOrg.get_owners(taxonomy).delete()
+    if org_owners:
+        TaxonomyOrg.objects.bulk_create(
+            [
+                TaxonomyOrg(
+                    taxonomy=taxonomy,
+                    org=org,
+                    rel_type=TaxonomyOrg.RelType.OWNER,
+                )
+                for org in org_owners
+            ]
+        )
+
+
+def get_taxonomies_for_org(
+    org_owner: Organization = None, enabled=True
+) -> Iterator[Taxonomy]:
+    """
+    Generates a list of the enabled Taxonomies owned by the given org, sorted by name.
+
+    If you want the disabled Taxonomies, pass enabled=False.
+    If you want all Taxonomies (both enabled and disabled), pass enabled=None.
+    """
+    taxonomies = oel_tagging.get_taxonomies(enabled=enabled)
+    for taxonomy in taxonomies.all():
+        org_short_name = org_owner.short_name if org_owner else None
+        if TaxonomyOrg.is_owner(taxonomy, org_short_name):
+            yield taxonomy.cast()
+
+
+def get_object_tags(
+    object_id: str, taxonomy: Taxonomy = None, valid_only=True
+) -> Iterator[ContentTag]:
+    """
+    Generates a list of content tags for a given object.
+
+    Pass taxonomy to limit the returned object_tags to a specific taxonomy.
+
+    Pass valid_only=False when displaying tags to content authors, so they can see invalid tags too.
+    Invalid tags will (probably) be hidden from learners.
+    """
+    for object_tag in oel_tagging.get_object_tags(
+        object_id=object_id,
+        taxonomy=taxonomy,
+        valid_only=valid_only,
+    ):
+        yield ContentTag.cast(object_tag)
+
+
+def tag_object(
```

Comment:
Maybe the purpose of this wrapper would be more clear if it was named `tag_content_object` ?

Reply:
Addressed with https://github.com/openedx/edx-platform/pull/32661/commits/ef6d2f9055a8e3bb8a835b6dc0386c41d7420817#diff-0ed9e2727b75df5366745749e3b39718ffbefe4b60289d80feab3b0019feb66bL102-R104.

Also [renamed `get_object_tags` to `get_content_tags`](https://github.com/openedx/edx-platform/pull/32661/commits/ef6d2f9055a8e3bb8a835b6dc0386c41d7420817#diff-0ed9e2727b75df5366745749e3b39718ffbefe4b60289d80feab3b0019feb66bL83-R85).